src/js/structs/DSLFilter.ts

```diff
@@ -0,0 +1,29 @@
+import { FilterType } from "#SRC/js/constants/DSLFilterTypes";
+import List from "#SRC/js/structs/List";
+import Node from "#SRC/js/structs/Node";
+
+type Arguments = {
+  label: string;
+  text: string;
+};
+
+export default interface DSLFilter {
+  /**
+   * Returns true if the filter can be applied to the given operand
+   */
+  filterCanHandle(type: FilterType, args: Arguments): boolean;
+
+  /**
+   * Apply the filter to the given result set, using the operand details given
+   * Returns the new, filtered list
+   */
+  filterApply(
+    resultset: List<Node>,
+    type: FilterType,
+    args: Arguments
+  ): List<Node>;
+}
+
+// we need this as long as you find something when grep'ing for `extends DSLFilter`.
+// it's a backwards-compatibility-measure on our way of converting filters to TS.
+export default class DSLFilter {}
```

Comment:
2 default exports in this file?

Reply:
yes, one for the new type and one for an actual implementation.

Reply:
Oh ok, thanks for the explanation. I wonder would this work to make it clear which is actually exported?
```
type Bla = string 
export const Bla = {} 
```

Reply:
This would export the constant but not the type. I'm explicitly seeking to export a type with the same name, so type-signatures read good und imports are not as noisy.